### ISSUE TITLE
compile failed on Win10 with vs2019  v 14.20

### ISSUE THREAD

**Issue by L1nY4n** [TattdCodeMonkey/crc#30]
Hello, I'm try to compile it on Windows 10 using visual studio 2019 pre , Program Maintenance Utility Version is 14.20.27607.1 , something wrong with " cannot open file '..\priv\crc_nif.dll'" and no dll file was generated in floder , logs detail as follows

``` Microsoft (R) Program Maintenance Utility Version 14.20.27607.1
Copyright (C) Microsoft Corporation.  All rights reserved.

        echo # Auto-generated as part of Makefile.win, do not modify. > Makefile.auto.win
        erl -eval "io:format(\"~s~n\", [lists:concat([\"ERTS_INCLUDE_PATH=\", code:root_dir(), \"/erts-\", erlang:system_info(version), \"/include\"])])" -s init stop -noshell >> Makefile.auto.win
        nmake /                   /F Makefile.win app

Microsoft (R) Program Maintenance Utility Version 14.20.27607.1
Copyright (C) Microsoft Corporation.  All rights reserved.

        cl /O2 /EHsc /I"c:/Program Files/erl10.1/erts-10.1/include" /LD /MD /Fe"..\priv\crc_nif.dll"  nif\xnif_slice.c nif\checksum_xor.c nif\crc_8.c nif\crc_nif.c nif\crc_algorithm.c nif\crc_model.c nif\crc_resource.c
                      
Microsoft (R) C/C++ Optimizing Compiler Version 19.20.27607.1 for x86
Copyright (C) Microsoft Corporation.  All rights reserved.

xnif_slice.c
checksum_xor.c
crc_8.c
nif\crc_8.c(37): warning C4333: '>>': right shift by too large amount, data loss
crc_nif.c
crc_algorithm.c
c:\myprojectpath\deps\crc\c_src\nif\crc_algorithm_fast.c.h(60): warning C4333: '>>': right shift by too large amount, data loss                                                                                     crc_model.c
crc_resource.c
Generating Code...

Microsoft (R) Incremental Linker Version 14.20.27607.1
Copyright (C) Microsoft Corporation.  All rights reserved.

/dll
/implib:..\priv\crc_nif.lib
/out:..\priv\crc_nif.dll
xnif_slice.obj
checksum_xor.obj
crc_8.obj
crc_nif.obj
crc_algorithm.obj
crc_model.obj
crc_resource.obj
LINK : fatal error LNK1104: cannot open file '..\priv\crc_nif.dll'
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.20.27607\bin\HostX86\x86\cl.EXE"' : return code '0x2'                                                                                 Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.20.27607\bin\HostX86\x86\nmake.EXE"' : return code '0x2'
Stop. 
```
What should I do to fix this ?

**Comment by TattdCodeMonkey** [comment]
Do you know if it works in current (older) versions of Visual Studio? 

I don't currently have a Windows machine or Visual Studio license so I wouldn't be able to test & reproduce this. 

The error is likely coming from the [c_src/Makefile.win](https://github.com/TattdCodeMonkey/crc/blob/master/c_src/Makefile.win) if you wanted to try and debug that. I would just want to confirm any changes don't break with older versions of Visual Studio.

**Comment by L1nY4n** [comment]
Thank you for your reply!
I got the same error in vs2017 too, It seems something wrong with my c++ buildtool or env on Windows OS 
I would like to try older versions of VS or other way .

**Comment by igaryhe** [comment]
@L1nY4n Hi, I'm running into the exact same problem. Do you have the problem fixed or not? Would like to know your result

**Comment by L1nY4n** [comment]
@igaryhe  I didn't fix the issure at that time (3 yr ago ) and I implemente  a pure elixir version  for my case (crc16 only) , and I  make a test again  in `VS 2022` today and the problem still there.  the issure is exactly  `priv` not exist !  I have send a [PR](https://github.com/TattdCodeMonkey/crc/pull/45)  to fix this 😂  @TattdCodeMonkey

**Comment by TattdCodeMonkey** [comment]
Thanks for the PR @L1nY4n 

merged and released in v0.10.4
